p2p/pex/known_address.go

```diff
@@ -54,6 +55,14 @@ func (ka *knownAddress) markGood() {
 	ka.LastSuccess = now
 }
 
+func (ka *knownAddress) ban(banTime time.Duration) {
+	ka.LastBanTime = time.Now().Add(banTime)
```

Comment:
I think this should only extend the ban, not shorten it. For example, if someone banned the address until 16:00, and then someone else wants to ban them until 13:00, they should remain banned until 16:00. This will matter when we start having different ban periods for various offences.